lib/pbench/server/api/resources/datasets_contents.py

```diff
@@ -0,0 +1,160 @@
+from http import HTTPStatus
+from pathlib import Path
+
+from flask import current_app, jsonify
+from flask.wrappers import Request, Response
+
+from pbench.server import OperationCode, PbenchServerConfig
+from pbench.server.api.resources import (
+    APIAbort,
+    ApiAuthorizationType,
+    ApiBase,
+    ApiContext,
+    APIInternalError,
+    ApiMethod,
+    ApiParams,
+    ApiSchema,
+    Parameter,
+    ParamType,
+    Schema,
+)
+from pbench.server.cache_manager import (
+    BadDirpath,
+    CacheExtractBadPath,
+    CacheManager,
+    CacheObject,
+    CacheType,
+    TarballNotFound,
+)
+from pbench.server.database.models.datasets import Dataset
+
+
+class DatasetsContents(ApiBase):
+    """
+    API class to retrieve inventory files from a dataset
+    """
+
+    def __init__(self, config: PbenchServerConfig):
+        super().__init__(
+            config,
+            ApiSchema(
+                ApiMethod.GET,
+                OperationCode.READ,
+                uri_schema=Schema(
+                    Parameter("dataset", ParamType.DATASET, required=True),
+                    Parameter("target", ParamType.STRING, required=False),
+                ),
+                authorization=ApiAuthorizationType.DATASET,
+            ),
+        )
+
+    def _get(
+        self, params: ApiParams, request: Request, context: ApiContext
+    ) -> Response:
+        """
+        Returns metadata about the target file path within a tarball.
+
+        Args:
+            params: includes the uri parameters, which provide the dataset and target.
+            request: Original incoming Request object
+            context: API context dictionary
+
+        Raises:
+            APIAbort, reporting either "NOT_FOUND" or "UNSUPPORTED_MEDIA_TYPE"
+
+        GET /api/v1/datasets/{dataset}/contents/{target}
+        """
+
+        dataset: Dataset = params.uri["dataset"]
+        target = params.uri.get("target")
+        path = Path("" if target in ("/", None) else target)
```

Comment:
I was really bothered by `location` values of `"."` in the map for the root directory, and went to (perhaps too much) effort to stamp them out.

But, yeah; `Path("")` isn't just "equivalent to" `Path(".")` ... the `str()` of `Path("")` *is* `"."`. I got sucked into all this as a consequence of "invisibilizing" the self-named root directory and building the cache map from there, while really trying to avoid "." on the output paths.